### PR TITLE
New UI and work without MCP server embedded

### PR DESCRIPTION
**Pull request by msanatan**
We feature a new UI built with UI Toolkit

- It's slightly nicer on the eyes, and works well in dark and light modes.
- The UX is intentionally more explicit than implicit. For e.g. the current window handles a lot of edge cases, this one doesn't. If a user wants to override a path, they can do so explicitly. It makes it easier to test and debug, and the code is more maintainable
- We can open the window with Cmd/Ctrl+Shift+M 
- If a user installs this from the Unity Asset Store, they have an option to download the MCP server, whereas Git users have the option to Rebuild it with their changes. 
- CI/CD uploads the server to the release as well

## Summary by CodeRabbit

- **New Features**
  - New MCP-for-Unity editor window with Settings, Connection, and Client sections; per-client configuration, register/unregister flows, and bridge health check (ping/pong).
  - In-app server download/install, version detection, and automated install/manage actions.

- **Improvements**
  - Service layer for bridge, client configuration, and path resolution with override support; service locator for easier access.
  - Debounced script-refresh scheduling, Claude CLI integration, and improved telemetry/log routing.
  
- **Bug Fixes**
  - Graceful handling when embedded server is missing and clearer installer error paths.

- **Chores**
  - CI now attaches a packaged server ZIP to releases.

- **Documentation**
  - v6 UI and service-layer docs added.

### CODE REVIEW

File: .github/workflows/bump-version.yml [hunk 1]
```diff
@@ -108,3 +108,30 @@ jobs:
...
+        shell: bash
+        run: |
+          set -euo pipefail
+          TAG="v${NEW_VERSION}"
+
+          # Create release
+          gh release create "$TAG" \
+            --title "v${NEW_VERSION}" \
+            --notes "Release v${NEW_VERSION}" \
+            "mcp-for-unity-server-v${NEW_VERSION}.zip#MCP Server v${NEW_VERSION}"
```
[the remark this answers is not in the document]
Comment (msanatan) [reply .github/workflows/bump-version.yml:137]: Probably not an issue... if a tag was created and needed to be deleted, we would delete the tag and remove the bump commit before running this again

File: MCPForUnity/Editor/Helpers/ServerInstaller.cs [hunk 2]
```diff
@@ -696,5 +709,133 @@ private static bool ValidateUvBinary(string uvPath)
...
+
+                try
+                {
+                    ZipFile.ExtractToDirectory(tempZip, tempExtractDir);
+
+                    // The ZIP contains UnityMcpServer~ folder, find it and move its contents
+                    string extractedServerFolder = Path.Combine(tempExtractDir, "UnityMcpServer~");
+                    Directory.CreateDirectory(destRoot);
+                    CopyDirectoryRecursive(extractedServerFolder, destRoot);
+                }
```
[the remark this answers is not in the document]
Comment (msanatan) [reply MCPForUnity/Editor/Helpers/ServerInstaller.cs:770]: Not that high risk, we control what's being zipped in the GitHub Actions after all

File: MCPForUnity/Editor/Windows/MCPForUnityEditorWindowNew.cs [hunk 3]
```diff
@@ -0,0 +1,842 @@
...
+            {
+                UpdateServerStatusBanner();
+                UpdatePathOverrides();
+                EditorUtility.DisplayDialog(
+                    "Download Complete",
+                    "Server installed successfully! The bridge will start automatically.",
+                    "OK"
+                );
+            }
+        }
```
[the remark this answers is not in the document]
Comment (msanatan) [reply MCPForUnity/Editor/Windows/MCPForUnityEditorWindowNew.cs:598]: We'll change the message to fix

File: MCPForUnity/Editor/Windows/MCPForUnityEditorWindowNew.cs [hunk 4]
```diff
@@ -0,0 +1,842 @@
...
+
+            // Get config JSON
+            string configJson = MCPServiceLocator.Client.GenerateConfigJson(client);
+            configJsonField.value = configJson;
+
+            // Get installation steps
+            string steps = MCPServiceLocator.Client.GetInstallationSteps(client);
+            installationStepsLabel.text = steps;
+        }
+
```
[the remark this answers is not in the document]
Comment (msanatan) [reply MCPForUnity/Editor/Windows/MCPForUnityEditorWindowNew.cs:433]: This should be resolved now, I reverted to UnityMCP so nothing breaks with people who upgrade

File: MCPForUnity/Editor/Services/ClientConfigurationService.cs [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+                {
+                    claudeClient.SetStatus(McpStatus.NotConfigured);
+                }
+                Debug.Log("<b><color=#2EA3FF>MCP-FOR-UNITY</color></b>: No MCP for Unity server found - already unregistered.");
+                return;
+            }
+
+            // Remove the server
+            if (ExecPath.TryRun(claudePath, "mcp remove unityMCP", projectDir, out var stdout, out var stderr, 10000, pathPrepend))
+            {
```
[the remark this answers is not in the document]
Comment (msanatan) [reply MCPForUnity/Editor/Services/ClientConfigurationService.cs:332]: https://github.com/CoplayDev/unity-mcp/pull/313#discussion_r2422460679

File: docs/v6_NEW_UI_CHANGES.md [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+// Access path resolver service
+string mcpServerPath = MCPServiceLocator.Paths.GetMcpServerPath();
+```
+
+**Benefits:**
+- No constructor dependencies (easy to use anywhere)
+- Lazy initialization (services created only when needed)
+- Testable (supports custom implementations via `Register()`)
+
```
[the remark this answers is not in the document]
Comment (msanatan) [reply docs/v6_NEW_UI_CHANGES.md:114]:
```csharp
namespace MCPForUnity.Editor.Services
{
    /// <summary>
    /// Service locator for accessing MCP services without dependency injection
    /// </summary>
    public static class MCPServiceLocator
    {
        private static IBridgeControlService _bridgeService;
        private static IClientConfigurationService _clientService;
        private static IPathResolverService _pathService;

        /// <summary>
        /// Gets the bridge control service
        /// </summary>
        public static IBridgeControlService Bridge => _bridgeService ??= new BridgeControlService();

        /// <summary>
        /// Gets the client configuration service
        /// </summary>
        public static IClientConfigurationService Client => _clientService ??= new ClientConfigurationService();

        /// <summary>
        /// Gets the path resolver service
        /// </summary>
        public static IPathResolverService Paths => _pathService ??= new PathResolverService();

        /// <summary>
        /// Registers a custom implementation for a service (useful for testing)
        /// </summary>
        /// <typeparam name="T">The service interface type</typeparam>
        /// <param name="implementation">The implementation to register</param>
        public static void Register<T>(T implementation) where T : class
        {
            if (implementation is IBridgeControlService b)
                _bridgeService = b;
            else if (implementation is IClientConfigurationService c)
                _clientService = c;
            else if (implementation is IPathResolverService p)
                _pathService = p;
        }

        /// <summary>
        /// Resets all services to their default implementations (useful for testing)
        /// </summary>
        public static void Reset()
        {
            _bridgeService = null;
            _clientService = null;
            _pathService = null;
        }
    }
}

```

It does expose a `Register` function
